### PR TITLE
New quality checks

### PR DESCRIPTION
**Pull request by StefanVriend**

### CODE REVIEW

**Comment by StefanVriend** [comment]
Remaining tasks:

- Check unusual coordinates (see #39). Cluster the lat/long and throw warnings for cases outside of these clusters. Only warnings, because there may be unusual records (e.g. dead bird found outside of the study site).
- Check overlapping nests (see #40). Can use lubridate ‘overlap’ function.
- Update documentation (see #116).
- Change reference value checks to use data rather than arbitrary values (see #117).

**Comment by StefanVriend** [comment]
Reference values in Brood_data (see check_values_brood, from line 716 onwards) are now generated by the data.

If number of observations for a particular population-species combination is too low, I suggest to skip these records in the check rather than comparing against Hoge Veluwe/NIOO data. This comparison might be more or less valid for variables like clutch size, but is far from reasonable in the case of phenology-related variables, like laying date.

Reference value determination:

- Values for ClutchSize, BroodSize and NumberFledged are considered a warning if larger than the 99th percentile, and an error if negative or 4 times the 99th percentile. 

- Values for LayDate are considered a warning if earlier than the 1st percentile or later than the 99th percentile, and an error if earlier than Jan 1st or later than Dec 31st.

**Comment by LiamDBailey** [comment]
@StefanVriend not a bad idea. I think it might be better to hold off on assigning errors/warnings rather than giving poor errors/warnings that we'll then have to discuss with the data owners. Do you know how many populations currently would **_not_** be tested using this 50 observation cutoff?

File: R/brood_check.R
Comment (LiamDBailey) [inline R/brood_check.R:78]:
```suggestion
  # - Compare lay and hatch dates
```

File: R/create_reference_values.R
Comment (LiamDBailey) [inline R/create_reference_values.R:18]: Can now be removed?

**Comment by StefanVriend** [comment]
These population-species combinations do not make the cut:
```
   Species PopID     n
   <chr>   <chr> <int>
 1 POEPAL  BUU       1
 2 SITEUR  BUU      49
 3 POEPAL  HOG      27
 4 PERATE  LIE       1
 5 PERATE  MIS      10
 6 PERATE  MUR       1
 7 FICHYP  OOS      10
 8 PASMON  OOS       2
 9 POEPAL  OOS       1
10 SITEUR  OOS      15
11 PERATE  VLI      25
12 PERATE  WAR      40
13 SITEUR  WAR      10
14 POEPAL  WES       7
```

**Review by LiamDBailey — COMMENTED** [review]
Looks good, just a few minor suggestions. I think the idea to only use local data when sample size is large enough is fine, in which case we delete the code in `create_reference_values.R` (l.17+)

File: R/create_reference_values.R
Comment (StefanVriend) [reply R/create_reference_values.R:18]: The first part of that function, so the lines concerning brood data, can be removed, but the lines concerning capture data have to stay because the check_values_capture function has not been updated yet.

**Comment by StefanVriend** [comment]
Reference values in Capture_data (see `check_values_capture`, from line 245 onwards) are now generated by the data (see #117).

For adult mass, adult tarsus and chick tarsus records, we use the same threshold as with brood values: > 50 observations, reference values are created; < 50 check is skipped.

For chick mass, the function tries to fit a logistic growth model (for which I adjusted the code that @LiamDBailey made, `calculate_chick_mass_cutoffs`). If this fails, the function will continue and create reference values per age. Ages with < 50 observations are skipped.

Currently the threshold is set at 50 observations. However, we might miss an obvious outlier if there are 49 "correct" records and 1 "incorrect" record, if that is an obvious outlier (as it pulls the 99th percentile up). I consider raising this threshold to 100 instead. 

What do you think? @LiamDBailey, @ChloeRN, @zuzulaz

Once this PR is ready to be merged, I will check the conflicting files. I suspect the conflict with the internal data update PR that was merged a week ago.

File: R/brood_check.R
Comment (LiamDBailey) [inline R/brood_check.R:763]: Perhaps an easier approach would be to just find the difference between dates. That way you don't have to multiply by 365 (which could cause some weirdness with leapyears). Suggestion below, not much prettier but I think a bit more robust.

```suggestion
      dplyr::mutate(!!paste0(var, "_julian") := as.numeric(!!rlang::sym(var) - lubridate::ymd(paste(BreedingSeason, "1", "1", sep = "-")) + 1)) %>%
```

File: R/brood_check.R
Comment (LiamDBailey) [inline R/brood_check.R:888]: As above

File: R/brood_check.R
Comment (LiamDBailey) [inline R/brood_check.R:947]: As above

**Review by LiamDBailey — CHANGES_REQUESTED** [review]
Changes look fine but it's currently failing the tests (running `devtools::test()`). It fails at setup-xyz.R. Try and get this working before we merge.

**Comment by StefanVriend** [comment]
Thanks for the review. I'm trying to run the test, but there's an issue running the pipelines for the test. In line 115, when cycling through the directories something goes wrong. I have no clue why it fails as I have not changed anything in `run_pipelines`...

**Pull request merged**
